### ISSUE TITLE
Tunneling issue, redirecting to local IP

### ISSUE THREAD

**Issue by dguber-axle** [boringproxy/boringproxy#190]
Hello,

I'm having the following issue:
I try to access my web app threw the reverse proxy and the first request gets passed through okay.
The problem is that the second and following requests which bring some CSS files and JS files get rewritten their address to "127.0.0.1" and it breaks the tunneling.

Is this an issue of boringproxy or a configuration issue of my web server?

Thanks.

**Comment by coder0987** [comment]
Hello @dguber-axle ,
I believe it is a boringproxy issue since I am facing it as well (see [the Indie Bits forum](https://forum.indiebits.io/t/redirect-to-127-0-0-1-using-boringproxy-with-nextcloud-dokuwiki-collabora-online-revive-adserver-and-others/148/4) )

My current guess would be that boringproxy fails to set the Host header
This is reinforced by the Nextcloud Android app crashing with error "host parameter is null" when using boringproxy with Nextcloud (see [this error report](https://github.com/nextcloud/android/issues/10664) )

I was advised to set the tunnel to "Client Raw TLS" (which does maintain auto HTTPS) and it worked for DokuWiki but not for Nextcloud. Try testing that and see if it works for you

**Comment by dguber-axle** [comment]
Hey @coder0987 

Thanks, it looks like with Client Raw TLS it doesn't replace the URL but some stuff try to get loaded from HTTP and it isn't forcing them to be loaded from HTTPS.
So maybe I'll have to manually edit some scripts.

Thanks.

**Comment by coder0987** [comment]
No problem.
As a temporary fix, try using [HTTPS Everywhere](https://www.eff.org/https-everywhere)
I was having the same issue with it redirecting to HTTP pages. Let me know what you find out

**Comment by dguber-axle** [comment]
Great, it worked that way temporary! 
But I'd like to find a proper solution.

Thanks!

**Comment by anderspitman** [comment]
Yeah we've had issues with redirects in the past. Do you have a simple app that I can self-host with my own boringproxy to try and debug this?

**Comment by dguber-axle** [comment]
Hello @anderspitman,

I don't have a simple app that you can host, but we could get in contact via Skype or email and I can show you the issue on my set up if you'd like.

Thanks.

**Comment by mfitzgerald2** [comment]
I had the same issue with a WordPress site using a standard LAMP stack if that is any help, probably would be easier to fire up via Docker or in a VM

Seems like I could get it to go to the pages on the frontend through the TLS passthrough but it would freak out when I tried to hit /wp-admin

Happy to help any way I can

**Comment by coder0987** [comment]
You could try hosting [DokuWiki on docker](https://github.com/linuxserver/docker-dokuwiki)
Anytime you log in or save it redirects to localhost

**Comment by coder0987** [comment]
A note for anyone using DokuWiki:
There is a configuration option in the site settings described [[https://www.dokuwiki.org/config:baseurl|here]]
Go to https://wiki.example.com/doku.php?id=start&do=admin&page=config and set the baseurl value to https://wiki.example.com (no slash)
This seems to have solved the problem for DokuWiki, but Nextcloud for Android still is not working

**Comment by anderspitman** [comment]
Sorry for the long delay. Busy summer but a good one! I have more time for boringproxy now.

I've reviewed your reports and the code. I believe I understand now why `Client Raw TLS` partially solves the problem. Basically it causes the `Host` header to be passed through from the original request, whereas `Client HTTPS` and `Server HTTPS` always set the host to the IP address. I have a fix in for that if you wouldn't mind testing it:

https://gemdrive.io/apps/delver/?drive=https://files.apitman.com&path=/public/boringproxy/set-host-header

That said, `Client Raw TLS` should be properly redirecting HTTP to HTTPS. I just tested that to make sure. Do you have `-allow-http` enabled by chance?

**Comment by anderspitman** [comment]
I'm going to go ahead and merge this into master, since I'm fairly confident the fix is needed, even if there are other problems keeping this issue open.
